api/views.py

```diff
@@ -46,31 +52,78 @@ class ChildViewSet(FilterByUrlKwargsMixin, views.ModelViewSet):
+        """
+        studies = get_objects_for_user(self.request.user, 'studies.can_view_study_responses')
+        study_ids = studies.values_list('id', flat=True)
+        return Child.objects.filter(Q(response__study__id__in=study_ids) | Q(user__id=self.request.user.id)).distinct()
 
 
 class DemographicDataViewSet(FilterByUrlKwargsMixin, views.ModelViewSet):
     lookup_field = 'uuid'
     resource_name = 'demographics'
+    # TODO modify when oauth in
     queryset = DemographicData.objects.filter(user__is_active=True)
     serializer_class = DemographicDataSerializer
     filter_fields = [('user', 'user'), ]
+    http_method_names = [u'get', u'head', u'options']
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        """
+        Overrides queryset.
+
+        Shows 1) demographics attached to responses you can view, and 2) your own demographics
+        """
+        studies = get_objects_for_user(self.request.user, 'studies.can_view_study_responses')
+        study_ids = studies.values_list('id', flat=True)
+        return DemographicData.objects.filter(Q(response__study__id__in=study_ids) | Q(user__id=self.request.user.id)).distinct()
 
 
 class UserViewSet(FilterByUrlKwargsMixin, views.ModelViewSet):
     lookup_field = 'uuid'
     resource_name = 'users'
+    # TODO modify when oauth in
     queryset = User.objects.filter(demographics__isnull=False).distinct()
     serializer_class = UserSerializer
     filter_fields = [('child', 'children'), ('response', 'responses'), ]
+    http_method_names = [u'get', u'head', u'options']
+    permission_classes = [IsAuthenticated]
 
+    def get_queryset(self):
+        """
+        Overrides queryset.
+
+        Shows 1) users that have responded to studies you can view and 2) your own user object
+        """
+        # TODO should we further restrict on participants only?
+        studies = get_objects_for_user(self.request.user, 'studies.can_view_study_responses')
+        study_ids = studies.values_list('id', flat=True)
+        return User.objects.filter(Q(children__response__study__id__in=study_ids) | Q(id=self.request.user.id)).distinct()
 
 class StudyViewSet(FilterByUrlKwargsMixin, views.ModelViewSet):
     resource_name = 'studies'
     queryset = Study.objects.filter(state='active')
     serializer_class = StudySerializer
     lookup_field = 'uuid'
     filter_fields = [('response', 'responses'), ]
+    http_method_names = [u'get', u'head', u'options']
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        """
+        Shows studies that are either 1) active and public or 2) studies you have permission to edit.
 
+        "can_edit_study" permissions allows the researcher to preview the study before it has been made active/public
+        """
+        return (super().get_queryset() | get_objects_for_user(self.request.user, 'studies.can_edit_study')).distinct()
```

Comment:
I wasn't sure how to write this when I wanted to combine the queryset with using the guardian shortcut. Also, did you see this [question](https://github.com/CenterForOpenScience/lookit-api/pull/31#issuecomment-319964588)?